src/common/src/hash/dispatcher.rs

```diff
@@ -97,6 +99,7 @@ fn hash_key_size(data_type: &DataType) -> HashKeySize {
         DataType::Timestamp => HashKeySize::Fixed(size_of::<Timestamp>()),
         DataType::Timestamptz => HashKeySize::Fixed(size_of::<i64>()),
         DataType::Interval => HashKeySize::Fixed(size_of::<Interval>()),
+        DataType::Int256 => HashKeySize::Fixed(Int256::size()),
```

Comment:
When this returns `Fixed`, we need to impl `HashKeySerDe` or the following would panic:

```
select distinct v::int256 from generate_series(1, 1, 1) as t(v);
```

https://github.com/risingwavelabs/risingwave/blob/87f367b280d326f74df20dac54a050d2bc961b7e/src/common/src/hash/key.rs#L422-L432

Reply:
It seems that HashKeySerDe cannot be implemented for non-primitive types, because it needs to return a temporary scalar's scalar ref. However, int256 appears to be fixed-sized. 

Let me change it to Variable first